woocommerce/woocommerce
# Merge WCA install routines to the core

### All Submissions:






### Changes proposed in this Pull Request:

This PR merges WCA [Install.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Internal/Admin/Install.php) into the core [class-wc-install.php](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/class-wc-install.php)

Other than the following new changes, the rest is mainly copied and pasted.

**New changes:**
- Deprecated `woocommerce_admin_updated_existing`, `woocommerce_admin_newly_installed`, and `woocommerce_admin_updated` and added `woocommerce_newly_installed` and `woocommerce_updated_existing`. `woocommerce_admin_updated` is replaced by the existing `woocommerce_updated` action.
- Notes are deleted by using `$wpdb` directly as note stores are not initialized when the plugin is activated via CLI.



### How to test the changes in this Pull Request:

#### Installation

1. Start with a fresh install.
2. Navigate to the plugins page and activate the plugin
3. Check tables, corn jobs, actions, options, and notes.

#### Update

1. Start with a fresh install.
2. Lower `woocommerce_version` option value.
5. Navigate to any page.
6. Confirm the expected `$db_updates` are executed by visiting `WooCommerce -> Status -> Scheduled Actions`. You can also run plugins/woocommerce/tests/legacy/unit-tests/woocommerce-admin/install.php unit test.

#### Uninstall

1. Deactivate the plugin
2. Delete the plugin
3. Confirm the [scheduled actions](https://github.com/woocommerce/woocommerce/pull/32513/files#diff-7b4fefcc6400f2f0206479b1c1af7cc871a6e044a18329df20aae892605eb04eR26) are cleared.


WooCommerce and packaged WooCommerce Admin versions

| WooCommerce | WooCommerce Admin|
|----------------|------------------------|
| 4.0.0|1.0.0|
|4.4.0|1.4.0|
|4.5.0|1.5.0|
|5.0.0|1.9.0|
|5.6.0|2.5.1|
|6.0.0|2.9.4|
|6.3.0|3.2.1|


[] Plugin activation, deactivation, uninstall routines
[] Activation/deactivation via CLI (wp plugin activate/deactive woocommerce)
[] Activation/deactivation manually from the plugin page
[] Installs expected tables and corn jobs
[] Removes installed tables and cron jobs on uninstall
[] Fires expected [actions](https://github.com/woocommerce/woocommerce/pull/32513/files#diff-b082a396ea78d8aaa870e896edc388647bd548df1c4eddc9b4953df241faf99bR240)

### Testing Insutrctions

We have merged WooCommerce Admin install logics to WooCommerce. There are no visual changes involved, but it would be nice to run a few tests to make sure everything stays the same.

*Testing tables*

1. Start with a fresh WordPress
2. Install WooCommerce as usual
3. Go to `WooCommerce -> Status` and scroll down to the `Database` table.
4. Make sure the following tables exist.

wc_order_stats
wc_order_product_lookup
wc_order_tax_lookup
wc_order_coupon_lookup
wc_admin_notes
wc_admin_note_actions
wc_customer_lookup
wc_category_lookup

*Deactivation & Activation*

1. Install WooCommerce as usual
2. Go to `Plugins -> Installed Plugins`
3. Deactivate `WooCommerce` and confirm no errors.
4. Activate `WooCommerce` again and confirm no errors.
5.  Deactivate `WooCommerce` and then click `delete`. 

*Cron events*
1. Install WooCommerce as usual
2. Install [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/)
3. Go to `Tools -> Cron Events`
4. Search for `wc_admin_daily` and make sure it exists.

*Upgrading from lower versions*

We've been including WooCommerce Admin in WooCommerce since WooCommerce 4.0. We should confirm no errors occur when upgrading from the following situation

- Upgrade from WooComemrce 4.0 to WooCommerce 6.5
- Upgrade from WooComemrce 4.4.0 to WooCommerce 6.5
- Upgrade from WooComemrce 4.5.0 to WooCommerce 6.5
- Upgrade from WooComemrce 5.0.0 to WooCommerce 6.5
- Upgrade from WooComemrce 5.6.0 to WooCommerce 6.5
- Upgrade from WooComemrce 6.0.0 to WooCommerce 6.5
- Upgrade from WooComemrce 6.3..0 to WooCommerce 6.5



### Other information:




### Changelog entry

N/A

### FOR PR REVIEWER ONLY:
